src/System.Net.Sockets/tests/FunctionalTests/DualModeSocketTest.cs

```diff
@@ -2755,10 +2757,14 @@ protected void ReceiveFrom_Helper(IPAddress listenOn, IPAddress connectTo)
         {
             using (Socket serverSocket = new Socket(SocketType.Dgram, ProtocolType.Udp))
             {
-                serverSocket.ReceiveTimeout = 500;
+                serverSocket.ReceiveTimeout = 1000;
                 int port = serverSocket.BindToAnonymousPort(listenOn);
 
                 SocketUdpClient client = new SocketUdpClient(_log, serverSocket, connectTo, port);
+                if (!client._t.Wait(TimeSpan.FromSeconds(3)))
```

Comment:
In this case you could just Assert (ie., cause the test to fail if it's not completed after 3 sec) .. you can log a message in the assert.

Reply:
I was wondering about that but from prospective of this test that perhaps is not critical failure. 
But I wanted to leave some trace in place. I still have a plan to instrument tests with event tracing entries. In ideal case I would like tests to be quiet but I would like to have as much details as possible in case of failure.